### DISCUSSION TITLE
Need some help understanding `ForkFlowException`, please!

### DISCUSSION THREAD

**Discussion by moritzschmitz-oviva** [keycloak/keycloak#34945] [Q&A]
We have been seeing a few `ForkFlowException` in our logs.

It seems to be thrown here: https://github.com/keycloak/keycloak/blob/25.0.6/services/src/main/java/org/keycloak/authentication/DefaultAuthenticationFlow.java#L495. The lines around the code make it look as if this is expected.

What is a fork? There is also this explanation here: https://github.com/keycloak/keycloak/blob/25.0.6/server-spi-private/src/main/java/org/keycloak/authentication/FlowStatus.java#L65. However, I fail to understand it in the context of our application.

Also this explanation here: https://github.com/keycloak/keycloak/blob/25.0.6/server-spi-private/src/main/java/org/keycloak/authentication/AuthenticationFlowContext.java#L135.

Could perhaps someone give an example user flow of how I would end up with this exception?

**Comment by tillbrodbeck-oviva** [comment]
Now understood: https://github.com/keycloak/keycloak/issues/39970
